### ISSUE TITLE
MathML parsing does not conform with official MathML XML schema

### ISSUE THREAD

**Issue by opoudjis** [plurimath/plurimath#81]
There are 73 MathML errors in NIST SP 800-90B, and 12 of them, as far as I can tell, involve empty `<mrow/>`.

But as it turns out, there is nothing at all wrong with an empty mrow:

https://www.w3.org/Math/XMLSchema/mathml3/mathml3-presentation.xsd

````
<xs:element name="mrow" substitutionGroup="m:PresentationExpression">
<xs:complexType>
<xs:group minOccurs="0" maxOccurs="unbounded" ref="m:MathExpression"/>
<xs:attributeGroup ref="m:mrow.attributes"/>
</xs:complexType>
</xs:element>
````

It's very important that you NOT raise spurious syntax errors, since I am currently aborting on them (because if the MathML is invalid, we cannot process it.)

**Comment by ronaldtse** [comment]
@suleman-uzair the problem here is that the MathML parsing in Plurimath does not follow the official MathML XML schema, as linked above and here:

* https://www.w3.org/Math/XMLSchema/mathml3/mathml3-presentation.xsd

We need to switch MathML parsing to Ox ASAP, and parse things according to this schema. If the values are allowed by the schema, we need to allow that and operate on those properly.

**Comment by opoudjis** [comment]
The MathML of NIST SP 800-90B is now processing without complaint.

**Comment by suleman-uzair** [comment]
@ronaldtse We have switched to the Ox parser, do we need any other changes or should we close this issue?

**Comment by ronaldtse** [comment]
@suleman-uzair Other than Ox, we need to ensure that the generated output is compliant with the MathML XSD. This is best done by using rspec to validate the MathML output with the MathML schema.

**Comment by suleman-uzair** [comment]
> @suleman-uzair Other than Ox, we need to ensure that the generated output is compliant with the MathML XSD. This is best done by using rspec to validate the MathML output with the MathML schema.

@ronaldtse ,
We have added all the examples from [MathML v3](https://www.w3.org/TR/MathML3/#:~:text=Table%20of%20Contents-,1%20Introduction,-1.1%20Mathematics%20and) in Mathml spec's.

**Comment by ronaldtse** [comment]
@suleman-uzair if this is done can you close it with reference to the PR and the set of MathML tests? Thanks.

**Comment by suleman-uzair** [comment]
This issue's been resolved in **PR** **#92**, specs can be seen under **[v3 directory](https://github.com/plurimath/plurimath/tree/main/spec/plurimath/mathml/v3)**.

**Comment by opoudjis** [comment]
I have discovered this issue a bit too late :-/

AsciiMath and Latex indeed do not have universal coverage of Unicode, and that's just the way it is. The point of AsciiMath is to avoid Unicode, after all. We accept some degradation of quality in mapping from MathML to AsciiMath, and some loss of nuance — e.g. non-breaking spaces ending up as just normal spaces.

The absence of the degree sign in Asciimath is a slight surprise, but we cannot manufacture Asciimath markup that doesn't exist. If it isn't in https://github.com/asciidoctor/asciimath/blob/master/lib/asciimath/parser.rb (and our feedback has expanded that table already), then it isn't in use in the Asciimath community. We can inject Latex markup as @ronaldtse suggests, as readable pseudo-Asciimath — so long as there isn't an expectation of roundtripping. 

So yeah, I'm a bit uneasy about this direction taken, but I don't have any better ideas. We will need to document this of course.
